### PR TITLE
To add script for SoupX

### LINKED ISSUE #49: To add script for SoupX
## What are the scientific goals of the analysis?

To add script for SoupX.

### PR DESCRIPTION
**Pull request by AntoniaChroni**
# Purpose/implementation Section

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## What scientific question/module of analysis is your analysis addressing?

To add script for SoupX in the `upstream-analysis` module.

## What GitHub issue does your pull request address?

Closes #49.

## Directions for reviewers 
Tell potential reviewers what kind of feedback you are soliciting.

### Which areas should receive a particularly close look?

Please review for code logic and clarity.

## Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/upstream-analysis/02_run_SoupX.Rmd [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+  # Correcting expression profile and clean data
+  # This will contain a corrected matrix to be used in place of the original table of counts in downstream analyses.
+  out <- adjustCounts(sc, roundToInt = TRUE) # roundToInt option to make sure we output integer matrix
+  colnames(out) <- paste0(sample_name, ":", colnames(out))
+  colnames(out) = gsub('-1$','', colnames(out)) # remove -1
+
+  # Let's save corrected matrix as a separate assay
+  seurat_obj  <- CreateSeuratObject(counts = out)
+  seurat_obj[["RNA_SoupX"]] <- CreateAssayObject(counts = seurat_obj@assays$RNA@counts)
+  seurat_obj@assays$RNA@counts <- out
```
Comment (sharonfreshour) [inline analyses/upstream-analysis/02_run_SoupX.Rmd:271]: Mostly for my understanding, what is the reason for creating the Seurat object with counts = out, then resetting seurat_obj@assays$RNA@counts to out?

Comment (sharonfreshour) [reply analyses/upstream-analysis/02_run_SoupX.Rmd:271]: I see that the RNA_SoupX assay is the SoupX corrected matrix. But I don't follow how `seurat_obj@assays$RNA@counts <- out` is saving the raw matrix.

Doesn't the `out` matrix only have the SoupX corrected counts, based on how it is created by the code in line 264: `out <- adjustCounts(sc, roundToInt = TRUE)`? So `seurat_obj@assays$RNA@counts <- out` would also be saving the corrected matrix in the RNA assay?

Comment (AntoniaChroni) [reply analyses/upstream-analysis/02_run_SoupX.Rmd]: This code chunk has been removed now as it is taken care of in the `04_run_filter_object.Rmd` script. See fb4dace73b026de44aeaa80e8060b3611558e8ed.

**Review by sharonfreshour — APPROVED** [review]
Looks good!

**Pull request merged**